pkg/reconciler/instances/istio/action.go

```diff
@@ -41,5 +44,36 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
 		return errors.Wrap(err, "Could not patch MutatingWebhookConfiguration")
 	}
 
+	generated := generateNewManifestWithoutIstioOperatorFrom(manifest.Manifest)
+	fmt.Println(generated)
+	_, err = context.KubeClient.Deploy(context.Context, generated, istioNamespace, nil)
+	if err != nil {
+		return errors.Wrap(err, "Could not deploy Istio resources")
+	}
+
 	return nil
 }
+
+func generateNewManifestWithoutIstioOperatorFrom(manifest string) string {
+	unstructs, err := kubeclient.ToUnstructured([]byte(manifest), true)
+	if err != nil {
+		return ""
+	}
+
+	builder := strings.Builder{}
+	for _, unstruct := range unstructs {
+		if unstruct.GetKind() == "IstioOperator" {
+			continue
+		}
+
+		unstructBytes, err := unstruct.MarshalJSON()
+		if err != nil {
+			return ""
+		}
+
+		builder.WriteString("---\n")
```

Comment:
Would recommend not to serialize each `Unstructured` one by one and add YAML delimiters (`---`) manually to it. Better to rely completely on the the K8s API to convert it back to YAML.

Therefore, an option could be to drop just the element from the slice of `Unstructured` entities. Like

      var withoutOperator []unstructured.Unstructured
      for _, unstr := range unstructs {
         if unstruct.GetKind() == "IstioOperator" {
            continue
         }
         withoutOperator = append(withoutOperator, unstr)
      }
      
Finally, serialise it, e.g. by using `unstructured.UnstructuredJSONScheme.Encode()` ([link to API](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L347-L351))

Reply:
It's actually the opposite of this call: https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go#L91

Reply:
If you like, feel free to add this new function for converting `unstructured.Unstructured` to our API https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/kubernetes/kubeclient/unstructured.go ;)